gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/membership/domain_service/IntegrationPrimaryOwnerDomainService.java

```diff
@@ -34,8 +39,70 @@
 public class IntegrationPrimaryOwnerDomainService {
 
     private final MembershipCrudService membershipCrudService;
+    private final MembershipQueryService membershipQueryService;
+    private final UserCrudService userCrudService;
+    private final GroupQueryService groupQueryService;
     private final RoleQueryService roleQueryService;
 
+    public PrimaryOwnerEntity getIntegrationPrimaryOwner(final String organizationId, String integrationId)
+        throws IntegrationPrimaryOwnerNotFoundException {
+        return findPrimaryOwnerRole(organizationId)
+            .flatMap(role ->
+                findIntegrationPrimaryOwnerMembership(integrationId, role)
+                    .flatMap(membership ->
+                        switch (membership.getMemberType()) {
+                            case USER -> findUserPrimaryOwner(membership);
+                            case GROUP -> findGroupPrimaryOwner(membership, role.getId());
```

Comment:
With the new decision we can't have group primary owner. I suppose we can simplify that